macros/format_float_value.sql

```diff
@@ -0,0 +1,7 @@
+{% macro format_float_value(value) %}
+
+    {% if target.name != "prod" %} CAST({{ value }} as Float32)
+    {% else %} {{ value }}
+    {% endif %}
```

Comment:
why is this conditional?

Reply:
I didn't want to change the actual query output, but dbt does something weird with floats and had trouble comparing the data in the tests

Reply:
Is it the values in the seed tables that are causing problems? I wonder if there's something we can do to get them inserted correctly. 🤔

Reply:
No, it's something within the dbt code that does the comparison. No matter what I changed the seeded values to, it always thought they were wrong. It seems like dbt is casting values as a mixture of decimal and float as it runs which changes the end decimals in float64. The first 6 or so digits are the same, but the rest are different from anything I could reproduce.

Reply:
What I'm seeing is that:

- Initial grading event in `xapi_events_all.csv` and the table as a value of `0.9523809523809523`. That value seems to carry through the whole stream and is what's expected at the end. 
- `grading_events_expected.csv` has a value of `0.95238096` for that event
- `fact_learner_course_grade_expected.csv` / `fact_student_status_expected.csv` have a value of `0.9523809552192688`

I've been testing with removing the format_float_value macro on `fact_student_status` and if I update the expected value to match what's in `xapi_events_all/.csv` the tests pass. My best guess is that the values are getting lost as part of the process of exporting the expected csv values. What are you doing to generate those? 

It would probably be good to document that as well, for when we need to add more tests later.

Reply:
Here's what I did:

I have xapi_all_events_parsed in my local clickhouse.
Event_id b47f3b76-21a8-45ec-b219-f11ef05739df shows score of 0.9523809523809523

Run grading_events query against that table
`
select
    event_id,
    CAST(emission_time, 'DateTime') as emission_time,
    actor_id,
    object_id,
    course_key,
    org,
    verb_id,
    JSONExtractFloat(event, 'result', 'score', 'scaled')
    as scaled_score
from xapi_events_all_parsed
where
    verb_id in (
        'http://id.tincanapi.com/verb/earned',
        'https://w3id.org/xapi/acrossx/verbs/evaluated'
    )
    or (
        verb_id in (
            'http://adlnet.gov/expapi/verbs/passed',
            'http://adlnet.gov/expapi/verbs/failed'
        )
        and JSON_VALUE(event::String, '$.object.definition.type')
        = 'http://adlnet.gov/expapi/activities/course'
    )
`
Export table from Clickhouse
Open in VSCode
Save as expected csv
Still has correct value in csv
![image](https://github.com/user-attachments/assets/a2377dd6-343e-44e2-821a-a3f5adabe04b)

Removed float macro from grading_events
dbt seed full refresh
dbt run full refresh
dbt test 

Test fails with:
![image](https://github.com/user-attachments/assets/0743a45d-fd8a-4c40-90e3-ca51bf235ee3)

Reply:
I also tried changing the unit test query to
`select event_id, emission_time, actor_id, object_id, course_key, org, verb_id, cast(scaled_score as Float64) scaled_score from grading_events_expected`
but it made no difference, the incorrect value was still used for the test

Reply:
Which test is it that's failing? 

I had a second change left over from earlier attempts that seems necessary to fix `test_fact_student_status.sql`, and that was a yaml to force the course_grade to a Float64:


unit-test-seeds/grading/grading_seeds.yaml:

```
version: 2

seeds:
  - name: fact_student_status_expected
    config:
      column_types:
        course_grade: Float64
```

Reply:
I think we would need to re-export anything with wrong values in the expected files, then add the yaml to force any that are somehow being implicitly created as Float32. But I thiiiiiiink that's all we need to get around the macro?

Reply:
The grading_events test is failing in what I sent. Adding that yaml for grading_events still doesn't work. I agree that changing the values on the other tests expected data would work, but they all stem from grading_events which fails without the float macro

Reply:
Ahhhh I got it. You're right, we need the column_type specification. I was using course_grade and not scaled_score so it wasn't working for grading_events.

I'll update everything and remove the macro. Thanks for the help!!